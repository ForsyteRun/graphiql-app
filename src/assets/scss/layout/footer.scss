/* footer */

.footer {
  flex-shrink: 0;
  @include font($family: $font-family-title);

  &__content {
    @include flex($justify: space-between, $wrap: wrap);
    padding: 1em 0;
    gap: 10px 40px;
    color: $background;

    & > * {
      flex: 1;
    }

    @include media('l') {
      @include flexColumn($align: stretch);
      gap: 2px;
    }
  }

  &__svg {
    height: 3rem;
    fill: $background;
    transition: 0.25s;
    filter: invert(100%);

    &:hover {
      scale: 1.1;
    }
  }

  &__rss-link {
    text-align: left;

    svg {
      background: url('../svg/rs.svg') center / contain no-repeat;
      width: 8rem;
    }
  }

  &__gh-link svg {
    background: url('../svg/github.svg') center / contain no-repeat;
    width: 3rem;
  }

  &__year-link {
    text-align: right;
  }
}
