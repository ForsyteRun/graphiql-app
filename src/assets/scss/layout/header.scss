/* header */

.header {
  @include flex($justify: space-between);
  background: $background-dark;
  width: 100%;
  height: 8rem;
  gap: 20px;
  padding: 0 20px;

  @include media('s') {
    height: 5rem;
    flex-wrap: wrap;
    justify-content: end;
    padding: 0 10px;
    gap: 10px;
  }

  @include media('xxs') {
    gap: 0;
  }

  & > * {
    flex: 1;
  }

  &__logo {
    text-align: left;

    .logo {
      padding: 15px 0;
      min-width: 150px;
      max-width: 280px;
      position: relative;
      margin-top: 2px;

      @include media('xxs') {
        min-width: auto;
      }

      a {
        white-space: nowrap;
      }

      &__image {
        display: inline-block;
        max-height: 50px;
        max-width: 100%;
        width: auto;
        text-decoration: none;
        margin-top: -22px;

        @include media('xs') {
          max-height: 40px;
        }

        @include media('xxs') {
          max-height: 30px;
        }
      }

      &__span {
        @include font(
          $family: $font-family-title,
          $size: 1.6rem,
          $weight: bold,
          $color: $links
        );
        margin-left: 10px;

        @include media('xxs') {
          display: none;
        }
      }
    }
  }

  &__links {
    @include flex($justify: center);
    gap: 30px;
  }

  &__buttons {
    @include flex($justify: flex-end);

    &-icon {
      @include media('s') {
        @include flex($justify: flex-start, $align: baseline);
        position: relative;
        height: 35px;
        width: 35px;
        margin: 0;
        left: 0;
        top: 0;
        right: 0;
        background: url(/src/assets/svg/profile.svg) no-repeat center center /
          cover;
        cursor: pointer;
      }

      @include media('s') {
        height: 25px;
        width: 25px;
        margin-top: -15px;
      }
    }

    &-list {
      @include flex($justify: flex-end);
      gap: 10px;

      @include media('s') {
        display: none;
      }

      &.active {
        @include media('s') {
          position: absolute;
          @include flexColumn($align: flex-start);
          gap: 10px;
          margin-top: 72px;
          margin-right: -7px;
          background: #fad7ee;
          padding: 7px 14px;
          border-radius: 12px;
          box-shadow:
            0 0 0 1px #ffffff0a,
            0 5px 7px 0 #ffffff38;
          z-index: 1;
        }

        @include media('xs') {
          margin-top: 75px;
        }

        &::before,
        &::after {
          content: '';
          position: absolute;
          top: -12px;
          right: 7%;
          border-color: transparent;
          border-style: solid;
          border-width: 6.5px;
          border-bottom-color: #fad7ee;
        }

        &::before {
          top: -13px;
          border-bottom-color: rgba(0, 0, 0, 0.1);
        }
      }
    }
  }
}
