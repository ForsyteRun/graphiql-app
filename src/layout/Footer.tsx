import { WELCOME_ROUTE } from '../constants/route';

const Footer = () => {
  return (
    <footer className="footer container">
      <div className="footer__content">
        <a
          className="footer__rss-link"
          href="https://rs.school/react/"
          target="_blank"
          rel="noreferrer"
        >
          <svg className="footer__svg"></svg>
        </a>
        <a
          className="footer__gh-link"
          href="https://github.com/ForsyteRun/graphiql-app/"
          target="_blank"
          rel="noreferrer"
        >
          <svg className="footer__svg"></svg>
        </a>
        <div className="footer__year-link">
          <span>@Copyright {new Date().getFullYear()}</span>{' '}
          <a href={WELCOME_ROUTE}>GraphiQL</a>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
