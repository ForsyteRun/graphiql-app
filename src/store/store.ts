import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import userReducer from './slice/userSlice';

const reducer = combineReducers({
  user: userReducer,
});

const store = configureStore({
  reducer,
});

setupListeners(store.dispatch);

export default store;
