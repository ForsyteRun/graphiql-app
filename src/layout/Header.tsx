import { NavLink } from 'react-router-dom';
import { Logo } from '../components';
import {
  AUTH_ROUTE,
  MAIN_ROUTE,
  REGISTRATION_ROUTE,
  WELCOME_ROUTE,
} from '../constants/route';
import { LogOut } from '../components/Logout';
import { AuthState } from '../utils/AuthState';
import { useAppSelector } from '../store/types';

const Header = () => {
  AuthState();
  const { isAuth } = useAppSelector((state) => state.user);
  return (
    <header className="header container">
      <div className="header__logo">
        <Logo />
      </div>
      <div className="header__links">
        <NavLink to={WELCOME_ROUTE} className="link">
          Приветствие
        </NavLink>
        {isAuth ? (
          <>
            <NavLink to={MAIN_ROUTE} className="link">
              Редактор
            </NavLink>
            <LogOut />
          </>
        ) : null}
      </div>
      {!isAuth && (
        <div className="header__buttons">
          <NavLink to={REGISTRATION_ROUTE}>
            <button className="button button-second">Регистрация</button>
          </NavLink>
          <NavLink to={AUTH_ROUTE}>
            <button className="button">Вход</button>
          </NavLink>
        </div>
      )}
    </header>
  );
};

export default Header;
