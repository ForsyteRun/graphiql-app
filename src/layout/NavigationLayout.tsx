import { Outlet } from 'react-router-dom';
import Header from './Header';
import Footer from './Footer';

function NavigationLayout() {
  return (
    <div className="tygh">
      <Header />
      <main className="content">
        <Outlet />
      </main>
      <Footer />
    </div>
  );
}

export default NavigationLayout;
