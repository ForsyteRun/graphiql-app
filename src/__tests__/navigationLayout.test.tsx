import '@testing-library/jest-dom';
import { NavigationLayout } from '../layout';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store/store';
import { BrowserRouter } from 'react-router-dom';

describe('NavigationLayout component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavigationLayout />
        </BrowserRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
