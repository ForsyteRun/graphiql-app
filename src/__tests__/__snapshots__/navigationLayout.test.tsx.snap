// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLayout component should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="header container"
  >
    <div
      class="header__logo"
    >
      <div
        class="logo"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
          title=""
        >
          <img
            alt="logo"
            class="logo__image"
            src="test-file-stub"
            title=""
          />
          <span
            class="logo__span"
          >
            GraphiQL
          </span>
        </a>
      </div>
    </div>
    <div
      class="header__links"
    >
      <a
        aria-current="page"
        class="link active"
        href="/"
      >
        Приветствие
      </a>
    </div>
    <div
      class="header__buttons"
    >
      <span
        class="header__buttons-icon"
      />
      <div
        class="header__buttons-list "
      >
        <a
          class=""
          href="/registration"
        >
          <button
            class="button button-second"
          >
            Регистрация
          </button>
        </a>
        <a
          class=""
          href="/login"
        >
          <button
            class="button"
          >
            Вход
          </button>
        </a>
      </div>
    </div>
  </header>
  <main
    class="content"
  />
  <footer
    class="footer container"
  >
    <div
      class="footer__content"
    >
      <a
        class="footer__rss-link"
        href="https://rs.school/react/"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          class="footer__svg"
        />
      </a>
      <a
        class="footer__gh-link"
        href="https://github.com/ForsyteRun/graphiql-app/"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          class="footer__svg"
        />
      </a>
      <div
        class="footer__year-link"
      >
        <span>
          @Copyright 2023
        </span>
         
        <a
          href="/"
        >
          GraphiQL
        </a>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
